server/controllers/files.ts

```diff
@@ -0,0 +1,109 @@
+import config from 'config';
+import { Request, Response } from 'express';
+
+import { canSeeExpenseAttachments } from '../graphql/common/expenses';
+import { getSignedGetURL, parseS3Url } from '../lib/awsS3';
+import { Collective, Expense, ExpenseAttachedFile, ExpenseItem, UploadedFile } from '../models';
+import { SUPPORTED_FILE_TYPES_IMAGES } from '../models/UploadedFile';
+
+async function hasUploadedFilePermission(req: Request, uploadedFile: UploadedFile): Promise<boolean> {
+  const actualUrl = uploadedFile.getDataValue('url');
+  if (req.remoteUser.id !== uploadedFile.CreatedByUserId) {
```

Comment:
I don't get this condition; if the remote user is the uploader, always return false?

In case it was reversed, I don't think the opposite should be true either: if I submit an expense as the admin of a collective, then later get removed from the admins, I shouldn't have access to the file anymore. In general, we don't want to use `CreatedByUserId` for permissions checks.

Reply:
I got this check wrong when refactoring the validation under this function, its fixed now.